src/app/(frontend)/embeds/providers/page.tsx

```diff
@@ -0,0 +1,117 @@
+import { ProviderPreview } from '@/components/ProviderPreview'
+import {
+  Accordion,
+  AccordionContent,
+  AccordionItem,
+  AccordionTrigger,
+} from '@/components/ui/accordion'
+import { getStateLabel } from '@/fields/location/states'
+import type { Provider } from '@/payload-types'
+import config from '@/payload.config'
+import { createLoader, parseAsString, SearchParams } from 'nuqs/server'
+import { getPayload } from 'payload'
+
+export const dynamic = 'force-static'
+
+const providersSearchParams = {
+  backgroundColor: parseAsString,
+  title: parseAsString,
+}
+
+const loadSearchParams = createLoader(providersSearchParams)
+
+type ProvidersByState = { [state: string]: Provider[] }
+
+export default async function ProvidersEmbedPage({
+  searchParams,
+}: {
+  searchParams: Promise<SearchParams>
+}) {
+  const { backgroundColor, title } = await loadSearchParams(searchParams)
+
+  const payload = await getPayload({ config })
+
+  const result = await payload.find({
+    collection: 'providers',
+    where: {
+      _status: {
+        equals: 'published',
+      },
+    },
+    limit: 1000,
+    depth: 0,
+  })
+
+  const providers = result.docs as Provider[]
+
+  // Organize providers by state based on statesServiced
+  // providers can be in multiple states
+  const providersByState: ProvidersByState = {}
+
+  providers.forEach((provider) => {
+    if (provider.statesServiced && provider.statesServiced.length > 0) {
+      provider.statesServiced.forEach((state) => {
+        if (!providersByState[state]) {
+          providersByState[state] = []
+        }
+        providersByState[state].push(provider)
+      })
+    }
+  })
+
+  const states = Object.keys(providersByState).sort()
+
+  // Split states into two columns for vertical flow
+  const midpoint = Math.ceil(states.length / 2)
+  const leftColumnStates = states.slice(0, midpoint)
+  const rightColumnStates = states.slice(midpoint)
+
+  return (
+    <div style={backgroundColor ? { backgroundColor } : undefined}>
+      {title && (
+        <div className="mb-6">
+          <h1 className="text-2xl font-bold mb-2">{title}</h1>
+        </div>
+      )}
+      <div className="grid sm:grid-cols-2 gap-x-4">
+        <StatesAccordion states={leftColumnStates} providersByState={providersByState} />
+        <StatesAccordion states={rightColumnStates} providersByState={providersByState} />
```

Comment:
We can clean this up by using [`columns`](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/columns)

```suggestion
      <div className="columns-2 gap-x-4">
        <StatesAccordion states={states} providersByState={providersByState} />
```

Reply:
Uh, why can't we? 

Personally, I don't think this isn't a clean solution 😅 Just three lines of code to split the array and abstracting some shared code into a component. 

But if you really don't like it or think it's not performant or something, we could use a .css file or a style object to use css columns...

Reply:
I was just calling out there are still pain points between css and js styling. 

I think you came up with the right solution here. I used CSS columns for event info and it worked well, but wont work here with the Accordion toggle interaction. See example with CSS below

![Kapture 2025-11-16 at 14 34 01](https://github.com/user-attachments/assets/283fdc84-a9f1-41de-8b6e-7f90ae5bd6d4)

Just a review comment.. no changes needed.